src/pipdeptree/_discovery.py

```diff
@@ -55,11 +59,17 @@ def get_installed_distributions(
             seen_dists[normalized_name] = dist
             dists.append(dist)
             continue
-        already_seen_dists = first_seen_to_already_seen_dists_dict.setdefault(seen_dists[normalized_name], [])
-        already_seen_dists.append(dist)
-
-    if first_seen_to_already_seen_dists_dict:
-        render_duplicated_dist_metadata_text(first_seen_to_already_seen_dists_dict)
+        if warning_printer.should_warn():
+            already_seen_dists = first_seen_to_already_seen_dists_dict.setdefault(seen_dists[normalized_name], [])
+            already_seen_dists.append(dist)
+
+    should_print_warning = warning_printer.should_warn() and first_seen_to_already_seen_dists_dict
```

Comment:
Like `dist_name_to_invalid_reqs_dict` in dag.py, it looks like we only need to check if `first_seen_to_already_seen_dists_dict` is empty if you prefer to keep the if statement.